examples/air-purifier-app/air-purifier-common/src/air-purifier-manager.cpp

```diff
@@ -191,13 +194,74 @@ void AirPurifierManager::HandleFanControlAttributeChange(AttributeId attributeId
     }
 }
 
+void AirPurifierManager::HandleOnOff(AttributeId attributeId, uint8_t type, uint16_t size, uint8_t * value)
+{
+    if (attributeId != OnOff::Attributes::OnOff::Id)
+    {
+        return;
+    }
+    bool on = static_cast<bool>(*value);
+    uint8_t new_speed;
+    uint8_t new_percent;
+    if (on)
+    {
+        ChipLogProgress(NotSpecified, "======================== Handle ON");
+        // If either of these come back as NULL, that should mean the fan is operating in auto mode.
+        // I have no idea what that means for this case, so I'll just set them to high because this
+        // is just an example.
+        // In theory these should always be NULL together or not all all, so hopefully the last
+        // error is more theoretical than practical.
```

Comment:
Not sure what "last error" is....

Reply:
updated the comment - I meant the checks for percent.IsNull or speed.IsNull when the first check for both IsNull falls through. When I wrote the comment, that was the last check. Then the checks expanded and the comment did not.